www-servers/tomcat/tomcat-10.1.24-r2.ebuild

```diff
@@ -81,28 +84,30 @@ src_prepare() {
 		slf4j-api.jar=$(java-pkg_getjars --build-only slf4j-api)
 		wsdl4j-lib.jar=$(java-pkg_getjars wsdl4j)
 	EOF
+	if use test; then
+		echo "easymock.jar=$(java-pkg_getjars --build-only easymock-3.2)" \
+			>> build.properties || die "easymock"
+	fi
 
 	# For use of catalina.sh in netbeans
 	sed -i -e "/^# ----- Execute The Requested Command/ a\
 		CLASSPATH=\`java-config --with-dependencies --classpath ${PN}-${SLOT}\`" \
 		bin/catalina.sh || die
-}
 
-EANT_BUILD_TARGET="deploy"
-EANT_TEST_GENTOO_CLASSPATH="easymock-3.2"
-EANT_GENTOO_CLASSPATH_EXTRA="${S}/output/classes"
-EANT_EXTRA_ARGS="-Dversion=${PV}-gentoo -Dversion.number=${PV} -Dcompile.debug=false"
+	# stop scrambling the build.xml
+	touch "${T}/java-ant-2_src_configure-run"
```

Comment:
this is preventing the build.xml rewrite?

Reply:
i'm asking because we used to do it by setting `JAVA_PKG_BSFIX="off"`:
https://github.com/gentoo/gentoo/blob/master/eclass/java-ant-2.eclass#L152

Reply:
> this is preventing the build.xml rewrite?

it does. we already have it in the actual `lucene-2.4.1-r4` ebuild.

Reply:
well, then it should be done the way i mentioned.

Reply:
> well, then it should be done the way i mentioned.
```
# @ECLASS_VARIABLE: JAVA_PKG_BSFIX
# @DESCRIPTION:
# Should we attempt to 'fix' ant build files to include the source/target
# attributes when calling javac?
JAVA_PKG_BSFIX=${JAVA_PKG_BSFIX:-"on"}
```
what is the benefit of not stopping `java-ant_bsfix` generally?
using `touch "${T}/java-ant-2_src_configure-run"` would definitely (`&& return`) stop any BSFIX completely.
i'd even prefer not inheriting `java-ant-2.eclass` but that seems possible only after enhancing the `eant()` function.

Reply:
i consider `touch "${T}/java-ant-2_src_configure-run"` internal hack as you have to use internals of the eclass to prevent something, whereas `JAVA_PKG_BSFIX` is designated to handle this in a transparent and easy to use way. in other words, `JAVA_PKG_BSFIX` is for me part of the public "API" whereas touching some file is a private implementation. the effect is the same, but imo the robustness of the code is different.

Reply:
> i consider `touch "${T}/java-ant-2_src_configure-run"` internal hack as you have to use internals of the eclass to prevent something, whereas `JAVA_PKG_BSFIX` is designated to handle this in a transparent and easy to use way. in other words, `JAVA_PKG_BSFIX` is for me part of the public "API" whereas touching some file is a private implementation. the effect is the same, but imo the robustness of the code is different.

got it. so we better wait and work on an enhancement to the `eant()` function to not generally force inheriting `java-ant-2` but only when the ebuild uses some of the `*BSFIX*` eclass variables.